src/tools/edit_line_tool.cpp

```diff
@@ -429,6 +429,11 @@ int EditLineTool::updateDirtyRectImpl(QRectF& rect)
 	
 	selection_extent = QRectF();
 	map()->includeSelectionRect(selection_extent);
+
+	const auto frame_extension = .001 * cur_map_widget->getMapView()
+	                             ->pixelToLength(clickTolerance()); // in mm
+	selection_extent.adjust(-frame_extension, -frame_extension,
+	                        frame_extension, frame_extension);
 	
 	rectInclude(rect, selection_extent);
 	int pixel_border = show_object_points ? pointHandles().displayRadius() : 1;
```

Comment:
Instead of adjusting the selection rectangle, we may simple increase the returned pixel border - this uses the same unit clickTolerance(). The common rationale is: `selection_extent` is from the data, `pixel_border` is from the tool and view.
~~~
int pixel_border = show_object_points ? std::max(qRound(clickTolerance()), pointHandles().displayRadius()) : 1;
~~~

Additional remark:
If you adopt this change, you might 
either eliminate this variable and move the expression to the return statement, 
or start with `int pixel_border = 1;`, and put another assignment into the `if (show_objects_points)` block.

Reply:
Bonus: Probably the whole view change slot machinery becomes obsolete by moving to the pixel_border approach.

Reply:
As far as I can tell, it is the `selection_extent` rectangle which defines the purple rectangle around the object at `/data/src/oomapper/src/tools/edit_point_tool.cpp:623`. But there is a valuable point in your remark. The purple rectangle should be perhaps extended by pixel_border as well to give even more breathing room to the outer vertices.
![image](https://user-images.githubusercontent.com/5584406/80072464-8d1e4e80-8546-11ea-90c1-05bac4d95c3f.png)

Reply:
The code would be like this:
```
	const auto pixel_border = show_object_points ? pointHandles().displayRadius() : 1;
	const auto frame_extension = 0.001 * cur_map_widget->getMapView()
	                             ->pixelToLength(pixel_border + clickTolerance()); // in mm
	selection_extent.adjust(-frame_extension, -frame_extension,
	                        frame_extension, frame_extension);

	rectInclude(rect, selection_extent);
<snip>
	return 0; // selection_extent covers the border points
```
OK?

Reply:
The unconditional inclusion of `pixel_border` is not perfect either as in most cases the room expansion for point handles is unnecessary because sufficient space is ensured by the object renderables themselves. IOW, most objects are massive enough to cover the point handles as in the below example rendered *without* the patch.
![image](https://user-images.githubusercontent.com/5584406/80075408-f1431180-854a-11ea-9a0c-13f4d2242c74.png)

At this point, however, I think the analysis is going into absurd dimensions and that perfect is the enemy of good. So, the question now is whether or not to include `pixel_border` into the purple frame. I'd leave it out as `clickTolerance()` already gives a sufficient margin and flat line endpoints are essentially a corner case.

Reply:
I had only a few minutes to test and check, and I seem to have failed in two ways: get the broad picture of the surrounding code, and actually testing the modifications which I meant to test. I'm sorry for the confusion.

Reply:
No p
No problem. On my side, it took me half a day to verify that the conversion of click tolerance value is correct. This code is quite complex.

Reply:
> This code is quite complex.

Yes. It is relying too much on side effects. It didn't really expect updateDirtyRect to update the box size, but I think I see why it is done. We need to mix two domains, map coordinates (mm) and screen coordinates (px). I would like to try another implementation approach soon (and I wouldn't know where to start without your PR). I just cannot switch branches two often while I'm working on 1561 (vector data georeferencing).

Reply:
FWIW, the API is a pain to work with. I would have suggestions for rework and I'd be willing to implement them. I'm, however, unsure if such intrusive changes would be welcome.

> I just cannot switch branches two often...

`git worktree` or `git stash` might help with work on multiple branches. I'll wait for your remarks on the alternatives.

Reply:
> FWIW, the API is a pain to work with. I would have suggestions for rework and I'd be willing to implement them. I'm, however, unsure if such intrusive changes would be welcome.

In general, yes. However, we should dicuss the suggestions first. (My hesitating with this PR comes from the fact that it adds more signals/slots, and I feel like it should be doable with regular interfaces for these classes because they are designed to work together closely.)

> > I just cannot switch branches two often...
> 
> `git worktree` or `git stash` might help with work on multiple branches. 

The problem are the build directories, not the source directories. I build and test multiple configurations.